### DISCUSSION TITLE
[Firebase-authentication] Saml provider in Identity Platform via Firebase usage for mobile

### DISCUSSION THREAD

**Discussion by eleojasmil** [firebase/flutterfire#7626] [Q&A]
https://cloud.google.com/identity-platform/docs/web/saml#redirect

Current Situation.
We already have a working web app and it uses SAML for authentication.
it is from Microsoft Azure SAML
```
const provider = new firebase.auth.SAMLAuthProvider('saml.myProvider');
firebase.auth().signInWithPopup(provider)
  .then((result) => {
  })
  .catch((error) => {
    // Handle error.
  });
```

but we also need to use that firebase authentication in mobile.
how can I access that existing FirebaseUser in mobile?
```
FirebaseAuth.instance.currentUser.uid
```

**Comment by billy1380** [comment]
Seems like there is an `SAMLAuthProvider` but the SDK throws an exception calling `signInWithPopup`. It looks like there is no way in `convertPlatformAuthProvider` to convert to a native `AuthProvider`.

Does any one know if there is a way to work around this?
